### ISSUE TITLE
Porting to ESP-idf

### ISSUE THREAD

**Issue by ahmedelashouti** [s00500/ESPUI#187]
Hallo everyone,
have anyone tried to port this library to esp idf ? maybe we can help each other instead of starting from zero !

**Comment by iangray001** [comment]
Not to my knowledge. 

I intend to play around with the IDF at some point but I haven’t so far. It shouldn’t be too hard to port this library, but the main thing would be our dependence on the Async web socket library.

**Comment by slunk-dot** [comment]
I made a mostly working ( good enough for my purposes ) port to esp-idf. As well as removing the dependency on  ESPAsyncWebServer there is Strings and ArduinoJson to handle. The code I have isn't suitable for a general use case, but if someone wants it as a starting point I can share it.

**Comment by MartinMueller2003** [comment]
The latest changes I made use the latest IDF in my test builds. You do need to play with the LittleFS titles a bit but beyond that it works.

**Comment by WangYuxin-esp** [comment]
Hi, The latest IDF seems to support Async web. Can we use this implementation scheme for this UI framework?
https://github.com/espressif/esp-idf/tree/master/examples/protocols/http_server/async_handlers

**Comment by MartinMueller2003** [comment]
ESPUI already uses an async web server implementation. Are you asking if we can change the library to the idf supported version? Is there an upside / advantage to making this change?

**Comment by WangYuxin-esp** [comment]
Yes, some friends prefer to use C language for development under the IDF framework instead of the Arduino framework. If this UI framework can be used under IDF, some web-based solutions can be used for rapid integration development.
